packages/actor-scraper/camoufox-scraper/src/internals/crawler_setup.ts

```diff
@@ -182,14 +180,22 @@ export class CrawlerSetup implements CrawlerSetupOptions {
             return req;
         });
 
-        this.requestList = await RequestList.open(
-            'PLAYWRIGHT_SCRAPER',
-            startUrls,
-        );
-
         // RequestQueue
         this.requestQueue = await RequestQueueV2.open(this.requestQueueName);
 
+        const requests: Request[] = [];
+        for await (const request of await RequestList.open(null, startUrls)) {
+            if (
+                this.input.maxResultsPerCrawl > 0 &&
+                requests.length >= 1.5 * this.input.maxResultsPerCrawl
+            ) {
+                break;
+            }
+            requests.push(request);
```

Comment:
I'm thinking this (initializing all of the `Request` instances preemptively) could potentially cause some OOM issues (especially with Cheerio crawler, which users tend to run with smaller RAM allocation). We are basically reverting [this PR](https://github.com/apify/crawlee/pull/2466).

Either way, there are no breaking changes in the API. Let's release it - reverting is always an option :shrug:

Reply:
You raise a good point here.

Reply:
If we do this - https://github.com/apify/crawlee/issues/2980 - we can make this thingy more efficient.